examples/auto_totem.js

```diff
@@ -0,0 +1,24 @@
+/*
+ * This script will automaticly set if a totem is in the inventory the totem in the off-hand.
+ * It checks for a totem every secound.
+ */
+const mineflayer = require('mineflayer')
+
+if (process.argv.length < 4 || process.argv.length > 6) {
+  console.log('Usage : node echo.js <host> <port> [<name>] [<password>]')
+  process.exit(1)
+}
+
+const bot = mineflayer.createBot({
+  host: process.argv[2],
+  port: parseInt(process.argv[3]),
+  username: process.argv[4] ? process.argv[4] : 'totem',
+  password: process.argv[5]
+})
+
+setInterval(() => {
+  var totem = bot.inventory.findInventoryItem(449, null)
```

Comment:
You shouldn't use the numeric id, because it changes from one version to an other. Instead, you can get the id from mcData like this:
```javascript
bot.on('spawn', () => {
  const mcData = require('minecraft-data')(bot.version) // You know the version when the bot has spawned

  if (mcData.itemsByName.totem_of_undying) { // Some versions don't have the totem
    setInterval(() => {
      const totemId = mcData.itemsByName.totem_of_undying.id // Get the correct id
      const totem = bot.inventory.findInventoryItem(totemId, null)
      if (totem) {
        bot.equip(totem, 'off-hand')
      }
    }, 1000)
  }
})
```

Reply:
@Karang When does the Bot get the inventory? I think it wouldnt be a good idea to search for a totem if the bot still has no inventory recieved.

Reply:
I think i will use spawn